pkg/guacanalytics/toposort.go

```diff
@@ -22,58 +22,65 @@ import (
 	"github.com/Khan/genqlient/graphql"
 )
 
-// TODO: add tests
-func ToposortFromBfsNodeMap(ctx context.Context, gqlClient graphql.Client, nodeMap map[string]BfsNode) (map[int][]string, []string, error) {
-	frontiers := make(map[int][]string)
-	parentsMap, infoNodes := copyParents(nodeMap)
-	frontierLevel := 0
+// TopoSortFromBfsNodeMap sorts the nodes such that it returns a map of level -> list of nodeIDs at that level
+func TopoSortFromBfsNodeMap(ctx context.Context, gqlClient graphql.Client, nodeMap map[string]BfsNode) (map[int][]string, []string, error) {
+	sortedNodes := make(map[int][]string) // map of level -> list of nodeIDs at that level
+	parentsMap, childrensMap, infoNodes := copyParents(nodeMap)
+	// parentsMap: map of nodeID (child) -> list of parents in the form of a map
+	// childrensMap: map of nodeID (parent) -> list of children in the form of a list
+	bfsLevel := 0
 	numNodes := 0
 	totalNodes := len(parentsMap)
 
 	for numNodes < totalNodes {
 		foundIDs := make(map[string]bool)
-		for id, parentsList := range parentsMap {
-			if len(parentsList) == 0 || (parentsList[0] == "" && len(parentsList) == 1) {
-				frontiers[frontierLevel] = append(frontiers[frontierLevel], id)
-				foundIDs[id] = true
+		for id, pMap := range parentsMap {
```

Comment:
I think the fact that there is both a pMap and a parentsMap variable, it is kind of confusing from the naming to figure out what each represents.   Maybe it would be better to name pMap parents or parentsSet or something along those lines... not sure what the best way around it would be?

Reply:
I have renamed `pMap` to `p`. I didn’t rename it to `parents` because calling `parents.parents` would be even more confusing.